basics/arrays.md

```diff
@@ -0,0 +1,120 @@
+# Arrays
+
+In D gibt es zwei verschiedene Arten von Arrays: **statische** und **dynamische**
+Arrays. Bei Zugriffen auf die Elemente eines Arrays wird immer überprüft ob der Index
+innerhalb des Arrays liegt, außer wenn der Compiler sicherstellen kann, dass diese
+Überprüfung nicht nötig ist.
+Wenn der Index außerhalb des Arrays liegt, wird ein `RangeError` geworfen, der die
+Anwendung standardmäßig stoppt.
+Dieses Feature kann durch die Compiler-Option `-boundscheck=off` ausgeschaltet
+werden, wodurch Geschwindigkeitsverbesserungen auf Kosten von reduzierter Sicherheit
+erreicht werden können.
+
+#### Statische Arrays
+
+Statische Arrays werden auf dem Stack gespeichert, wenn sie innerhalb einer
+Funktion definiert sind. Andernfalls werden sie in den statischen Speicher gesetzt.
+Sie haben eine feste Länge, welche zur Compile-Zeit bekannt ist. Der Typ
+beinhaltet hierbei die Länge:
+
+    int[8] arr;
+
+Der Typ von `arr` ist `int[8]`. Die Länge gehört zum Typ und wird nicht wie in
+C/C++ nach dem Variablennamen geschrieben.
+
+#### Dynamische Arrays
+
+Dynamische Arrays werden auf dem Heap gespeichert und können während der
+Laufzeit vergrößert und verkleinert werden. Ein dynamisches Array kann mit `new`
+und der Länge erstellt werden:
+
```

Comment:
Dynamische Arrays können zur Laufzeit erzeugt bzw. in ihrer Größe geändert werden. Dies geschieht durch Nutzung des Heap als Speicherort. Ein dynamisches Array wird mithilfe des `new`-Ausdrucks und der Länge erstellt: